### ISSUE TITLE
[Bug]: Heavy lag on Classic Era

### ISSUE THREAD

**Issue by pkrasimirov** [raethkcj/RatingBuster#138]
### Describe the bug

Works for sure in WotLK but not in Classic Era.  Shows only partial info in tooltip and lags heavily.  Disabling Stat summary makes the lag go away.  When `/console scriptErrors 1` is used it shows many errors one after the other, for exmple:

```
Message: Interface\AddOns\StatLogic\StatLogic.lua:2789: attempt to index field 'bonusArmorItemEquipLoc' (a nil value)
Time: Fri Feb  3 18:30:31 2023
Count: 1
Stack: Interface\AddOns\StatLogic\StatLogic.lua:2789: attempt to index field 'bonusArmorItemEquipLoc' (a nil value)
[string "@Interface\AddOns\StatLogic\StatLogic.lua"]:2789: in function `GetArmorDistribution'
[string "@Interface\AddOns\StatLogic\StatLogic.lua"]:2407: in function <Interface\AddOns\StatLogic\StatLogic.lua:2397>
[string "@Interface\AddOns\StatLogic\StatLogic.lua"]:2551: in function `GetSum'
[string "@Interface\AddOns\RatingBuster\RatingBuster.lua"]:3553: in function `StatSummary'
[string "@Interface\AddOns\RatingBuster\RatingBuster.lua"]:1824: in function `handler'
[string "@Interface\AddOns\RatingBuster\libs\TipHooker-1.0\TipHooker-1.0.lua"]:156: in function <...ns\RatingBuster\libs\TipHooker-1.0\TipHooker-1.0.lua:150>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: in function `SetBagItem'
[string "@Interface\FrameXML\ContainerFrame.lua"]:1352: in function <Interface\FrameXML\ContainerFrame.lua:1323>
[string "=[C]"]: in function `ContainerFrameItemButton_OnEnter'
[string "*:OnEnter"]:1: in function <[string "*:OnEnter"]:1>

Locals: self = <table> {
 GetArmorDistribution = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2772
 GetRAPFromAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1901
 GetReductionFromArmor = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1490
 GetSum = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2421
 argCheck = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:77
 GetRAPPerAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1862
 GetStatMod = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1456
 TalentCacheExists = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1366
 tip = StatLogicTooltip {
 }
 GetClassIdOrName = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:372
 InvalidateEvent = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1293
 GetRatingIdOrName = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:484
 GetStatNameFromID = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:376
 StatModInfo = <table> {
 }
 AuraInfo = <table> {
 }
 RemoveEnchant = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2164
 GetBlockValuePerStr = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1704
 StatModTable = <table> {
 }
 GetNormalManaRegenFromSpi = <function> defined @Interface\AddOns\StatLogic\Vanilla_Logic.lua:44
 GetAPFromAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1824
 GetSpellCritFromInt = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2085
 GetDiff = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:3019
 GetAPPerAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1782
 GetGemID = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2297
 GetHealthRegenFromSpi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2144
 GetEffectFromRating = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1579
 StatTable = <table> {
 }
 AreColorsEqual = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2765
 PatternTest = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:3135
 Bench = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:3115
 GetAPFromStr = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1661
 GenericStatMap = <table> {
 }
 BuildGemmedTooltip = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2207
 RemoveGem = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2168
 GetDodgePerAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1970
 tipMiner = StatLogicMinerTooltip {
 }
 GetAPPerStr = <function> defined @Interface\AddOns\StatLogic\StatLogic.
 ```

and

```
Message: Interface\AddOns\StatLogic\StatLogic.lua:2789: attempt to index field 'bonusArmorItemEquipLoc' (a nil value)
Time: Fri Feb  3 18:44:10 2023
Count: 4
Stack: Interface\AddOns\StatLogic\StatLogic.lua:2789: attempt to index field 'bonusArmorItemEquipLoc' (a nil value)
[string "@Interface\AddOns\StatLogic\StatLogic.lua"]:2789: in function `GetArmorDistribution'
[string "@Interface\AddOns\StatLogic\StatLogic.lua"]:2407: in function <Interface\AddOns\StatLogic\StatLogic.lua:2397>
[string "@Interface\AddOns\StatLogic\StatLogic.lua"]:2551: in function `GetSum'
[string "@Interface\AddOns\RatingBuster\RatingBuster.lua"]:3553: in function `StatSummary'
[string "@Interface\AddOns\RatingBuster\RatingBuster.lua"]:1824: in function `handler'
[string "@Interface\AddOns\RatingBuster\libs\TipHooker-1.0\TipHooker-1.0.lua"]:156: in function <...ns\RatingBuster\libs\TipHooker-1.0\TipHooker-1.0.lua:150>
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: ?
[string "=[C]"]: in function `SetBagItem'
[string "@Interface\FrameXML\ContainerFrame.lua"]:1352: in function `UpdateTooltip'
[string "@Interface\FrameXML\GameTooltip.lua"]:429: in function <Interface\FrameXML\GameTooltip.lua:408>
[string "=[C]"]: ?

Locals: self = <table> {
 GetArmorDistribution = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2772
 GetRAPFromAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1901
 GetReductionFromArmor = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1490
 GetSum = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2421
 argCheck = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:77
 GetRAPPerAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1862
 GetStatMod = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1456
 TalentCacheExists = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1366
 tip = StatLogicTooltip {
 }
 GetClassIdOrName = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:372
 InvalidateEvent = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1293
 GetRatingIdOrName = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:484
 GetStatNameFromID = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:376
 StatModInfo = <table> {
 }
 AuraInfo = <table> {
 }
 RemoveEnchant = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2164
 GetBlockValuePerStr = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1704
 StatModTable = <table> {
 }
 GetNormalManaRegenFromSpi = <function> defined @Interface\AddOns\StatLogic\Vanilla_Logic.lua:44
 GetAPFromAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1824
 GetSpellCritFromInt = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2085
 GetDiff = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:3019
 GetAPPerAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1782
 GetGemID = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2297
 GetHealthRegenFromSpi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2144
 GetEffectFromRating = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1579
 StatTable = <table> {
 }
 AreColorsEqual = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2765
 PatternTest = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:3135
 Bench = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:3115
 GetAPFromStr = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1661
 GenericStatMap = <table> {
 }
 BuildGemmedTooltip = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2207
 RemoveGem = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:2168
 GetDodgePerAgi = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1970
 tipMiner = StatLogicMinerTooltip {
 }
 GetAPPerStr = <function> defined @Interface\AddOns\StatLogic\StatLogic.lua:1625
 GetEffectFromDefense
```

### Addon Source

WowUp

### Version

1.3.7

### Locale

English

### Talent Specialization

Discipline Priest

### Race

Dwarf

**Comment by raethkcj** [comment]
Thank you for the report! Should be fixed in 1.7.4

**Comment by pkrasimirov** [comment]
Thank you so much for the quick fix!  I edited my file locally with your change and everything works perfectly.
